server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessagePreviewListener.java

```diff
@@ -90,9 +94,18 @@ private void handleAddedEvent(Added addedEvent, MailboxSession session) throws M
     private void preComputePreview(List<MessageId> messageIds, MailboxSession session) throws MailboxException {
         messageIdManager.getMessages(messageIds, FetchGroupImpl.BODY_CONTENT, session)
             .stream()
-            .map(Throwing.function((MessageResult messageResult) ->
-                messageFullViewFactory.fromMessageResults(ImmutableList.of(messageResult))))
-            .forEach(message -> storePreview(message.getId(), message.getPreview()));
+            .map(Throwing.function((MessageResult messageResult) -> computePreview(messageResult)))
+            .forEach(message -> storePreview(message.getT1(), message.getT2()));
+    }
+
+    private Tuple2<MessageId, Preview> computePreview(MessageResult messageResult) throws MailboxException, IOException {
```

Comment:
the tuple construction is not the responsibility of this method, it's something that is usually done in the map/flatmap pipeline